feature: replace added

A1exCart3r opened the pull request:
Лаб. Раб. 1, Задание 1, вариант 4
https://github.com/alexey-malov/oop/tree/master/tasks/01

A1exCart3r committed:
feature: replace added

Лаб. Раб. 1, Задание 1, вариант 4
https://github.com/alexey-malov/oop/tree/master/tasks/01

RuprechtKraus commented on replace/Program.cs:
```diff
@@ -0,0 +1,48 @@
+    {
+        if (args.Length != 4) 
+        {
+            Console.WriteLine("Неверное количество аргументов, введите аргументы в формате \"replace.exe <input file> <output file> <search string> <replace string>\"");
```
Слишком длинная строка. У тебя в IDE должна быть полоса, показывающая максимально предпочтительную длину строки:
![image](https://github.com/A1exCart3r/oop-lab1/assets/43097128/0131ac19-e35c-4c9b-a493-d0dc5123c601)

Разбей эту строку на две

A1exCart3r replied:
разбил

RuprechtKraus commented on replace/Program.cs:
```diff
@@ -0,0 +1,48 @@
+        }
+
+        string inputFile = args[0];
+        string lineFromFile;
```
Определи там, где используется

A1exCart3r replied:
переместил

RuprechtKraus commented on replace/Program.cs:
```diff
@@ -0,0 +1,48 @@
+            {
+                while (!sr.EndOfStream)
+                {
+                    if(searchText.Length != 0) 
```
Зачем эта проверка постоянно тут? Ты в самом начале один раз можешь проверить на длину а потом просто скопировать файл

A1exCart3r replied:
перепутал местами, исправил

RuprechtKraus commented on replace/Program.cs:
```diff
@@ -0,0 +1,48 @@
+                    }
+                }
+            }
+            return 0;
```
В самый конец кода лучше return 0. Если тебе надо будет добавить туда код то тебе тут удалять придется. Да и вообще логичный конец программы это в самом конце ее кода

A1exCart3r replied:
переместил

RuprechtKraus commented:
Опять ты обошелся стандартной функцией. Любой может Replace сделать. Я же тебе скидывал видео лекции, где разрабатывали эту программу со студентами. Ты должен свой аналог Replace сделать

RuprechtKraus commented on replace/tests.bat:
На пустой файл тоже желательно

A1exCart3r replied:
добавлено

A1exCart3r committed:
feature: replace v2 added

Задание
https://github.com/alexey-malov/oop/tree/master/tasks/01#%D0%B2%D0%B0%D1%80%D0%B8%D0%B0%D0%BD%D1%82-4--replace--80-%D0%B1%D0%B0%D0%BB%D0%BB%D0%BE%D0%B2-
